src/axolotl/utils/distributed.py

```diff
@@ -90,9 +106,7 @@ def gather_scalar_from_all_ranks(fn, world_size=1):  # pylint: disable=invalid-n
     value_scalar = fn()
     if not is_distributed():
         return [value_scalar]
-    value_tensor = torch.tensor(
-        value_scalar, device=torch.cuda.current_device()
-    ).float()
+    value_tensor = torch.tensor(value_scalar, device=get_device_type()).float()
```

Comment:
Would there be a regression here? `get_device_type()` returns a string cpu/npu/cuda.. where as `torch.cuda.current_device()` return an int.

Could you instead, create a utility that returns `torch.cuda.current_device()` or `torch.npu.current_device()` , given the current device?

Reply:
When only an int value is returned, torch will set the device of the tensor as cuda by default. Maybe I' ll change it to get_device_type() + str(torch.xxx.current_device) to ensure the tensor will be put on current card.